src/GeositeFramework/plugins/layer_selector_v2/main.js

```diff
@@ -160,14 +179,47 @@ define([
                     }
                 }, this);
 
-                _.each(visibleLayerIds, function(layerIds, serviceUrl) {
+                _.each(visibleLayerIds, function(layerServiceIds, serviceUrl) {
                     var mapLayer = this.map.getLayer(serviceUrl);
-                    if (layerIds.length === 0) {
+                    if (layerServiceIds.length === 0) {
                         mapLayer.setVisibleLayers([-1]);
                     } else {
-                        mapLayer.setVisibleLayers(layerIds);
+                        mapLayer.setVisibleLayers(layerServiceIds);
                     }
                 }, this);
+
+                this._setOpacityForService(visibleLayerIds);
+            },
+
+            _setOpacityForService: function(visibleLayerIds) {
+                // VisibleLayerIds is a map of serviceUrls -> selected layer service Ids
+                _.each(visibleLayerIds, function(layerServiceIds, serviceUrl) {
+                    var mapLayer = this.map.getLayer(serviceUrl),
+                        self = this;
+
+                    if (layerServiceIds.length !== 0) {
+                        var optionsArray = [],
+                            selectedLayers = this.state.getSelectedLayersForService(serviceUrl);
+
+                            _.each(selectedLayers, function(layerId) {
+                                var layerOpacity = self.state.getLayerOpacity(layerId);
+
+                                if (layerOpacity) {
+                                    var drawingOption = new LayerDrawingOptions({
+                                        // 0 is totally opaque, 100 is 100% transparent.
+                                        // Opacity is stored as a decimal from 0 (transparent)
+                                        // to 1 (opaque) so we convert it and invert it here.
+                                        transparency: 100 - (layerOpacity * 100)
+                                    });
+
+                                    var layer = self.state.findLayer(layerId);
+                                    optionsArray[layer.getServiceId()] = drawingOption;
+                                }
+                            });
+
+                            mapLayer.setLayerDrawingOptions(optionsArray);
```

Comment:
Not sure. It doesn't seem useful to set it if the layers aren't visible, so I'll skip this step if that's the case.